### PR TITLE
New/player

### PR DESCRIPTION
**Pull request by totine**
Player and Players (wrapper for collection) Implementation

Player has name, points and character; can swap his character,

Players are implemented on simple Player array; class can inform about its state, check addition of new player and swap players. It is ready for implement more than two players.

### CODE REVIEW

File: src/main/java/akademia/ox/IncorrectCharacterException.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package akademia.ox;
+
+public class IncorrectCharacterException extends RuntimeException {
```
Comment (AdrianHarenczyk) [inline src/main/java/akademia/ox/IncorrectCharacterException.java]: In this case it is not needed to catch this exception, is it intentional?

File: src/main/java/akademia/ox/Player.java [hunk 2]
```diff
@@ -11,6 +11,17 @@ public Player(String name) {
         this.points = 0;
     }
 
+    public Player(String name, String character) {
+        this.name = name;
+        this.points = 0;
+        try {
+            this.character = GameCharacter.valueOf(character);
```
Comment (AdrianHarenczyk) [inline src/main/java/akademia/ox/Player.java]: Perhaps is it not so good to catch exception in constructor, maybe throw it else would be better?

File: src/test/java/akademia/ox/PlayersTest.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package akademia.ox;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class PlayersTest {
+    private Player playerOne = new Player("samplePlayerOne", "X");
```
Comment (AdrianHarenczyk) [inline src/test/java/akademia/ox/PlayersTest.java:7]: Initialization of fields can be done with @BeforeTest annotation, it is loose suggestion.

File: src/main/java/akademia/ox/Player.java [hunk 4]
```diff
@@ -1,13 +1,43 @@
...
+    public int showPoints() {
+        return points;
+    }
+
+    public void incrementPoints(int newPoints) {
+        if (newPoints >= 0) {
+            points += newPoints;
+        }
+        else {
+            throw new IllegalArgumentException("Points shouldn't be negative");
```
Comment (AdrianHarenczyk) [inline src/main/java/akademia/ox/Player.java:34]: In this method i would add throws IAException

File: src/main/java/akademia/ox/IncorrectCharacterException.java (hunk 1 above)
Comment (totine) [reply src/main/java/akademia/ox/IncorrectCharacterException.java]: I will fix it

File: src/main/java/akademia/ox/Player.java (hunk 2 above)
Comment (totine) [reply src/main/java/akademia/ox/Player.java]: ok

**Review by AdrianHarenczyk — COMMENTED** [review]
Just these changes I suggested.

**Comment by AdrianHarenczyk** [comment]
Merge request accepted.

**Pull request merged**